client/app/scripts/superdesk-publish/publish.js

```diff
@@ -223,20 +260,91 @@
         };
 
         $scope.cancelSchedule = function(item) {
-            api.publish_queue.save(item, {'state': 'canceled', 'error_message': 'canceled by user'}).then(
-                function(response) {
-                    $scope.reload();
-                },
-                function(response) {
-                    if (angular.isDefined(response.data._issues)) {
-                        if (angular.isDefined(response.data._issues['validator exception'])) {
-                            notify.error(gettext('Error: ' + response.data._issues['validator exception']));
+            var _updates = {'state': 'canceled', 'error_message': 'canceled by user'};
+
+            if (angular.isDefined(item)) {
+                api.publish_queue.save(item, _updates).then(
+                    function(response) {
+                    },
+                    function(response) {
+                        if (angular.isDefined(response.data._issues)) {
+                            if (angular.isDefined(response.data._issues['validator exception'])) {
+                                notify.error(gettext('Error: ' + response.data._issues['validator exception']));
+                            }
```

Comment:
True. But those are raised by cerberus based on the validators declared in schema definition. Since `publish_queue` doesn't have validators declared at schema level, I didn't raise notifications.